sn_client/src/wallet.rs

```diff
@@ -64,6 +69,31 @@ impl WalletClient {
         }
     }
 
+    /// Send tokens to nodes closest to the data we want to make storage payment for.
+    pub async fn pay_for_storage(
+        &mut self,
+        content_addrs: impl Iterator<Item = &NetworkAddress>,
+    ) -> Result<(Dbc, PaymentProofsMap)> {
+        // TODO: calculate the amount to pay to each node, perhaps just 1 nano to begin with.
+        let amount = Token::from_nano(1);
+
+        // FIXME: calculate closest nodes to pay for storage, we are now just burning the output amount.
+        let to = vec![(amount, PublicAddress::new(SecretKey::random().public_key()))];
+
+        // Let's build the payment proofs for list of content addresses
+        let (reason_hash, payment_proofs) = build_payment_proofs(content_addrs)
```

Comment:
hmm...   I am bit lost here.
The function is `pay_for_storage` , which, as I understand, the `payment_proofs` shall be:
1, granted by the network nodes
OR
2, deduced from sender's balance

However, it seems the `build_payment_proofs` and `local_send` are all `local operations` which has no network actions involved.
Meanwhile, when the network nodes received the `payment_proofs` together with the chunk to be stored, it can only validate `the proof is related to that chunk` , but, how they can validate the `payment proof` is a valid one, so that they can accept the store?

Reply:
I've added a comment in the description, this is just a first step, subsequent PRs will bring in additional payment verification/validations.